// @bunkservices/bunk-api-lib/index.js
const axios = require('axios')

exports.meme = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/meme/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.donkey = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/donkey/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.subreddit = function(userID, token, subreddit) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/subreddit/${subreddit}/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.animeWall = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/anime-wallpaper/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.llama = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/llama/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.alpaca = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/alpaca/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.giraffe = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/giraffe/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.potato = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/potato/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.moose = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/moose/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}


exports.aww = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/aww/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.camel = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/camel/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.seal = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/seal/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.zebra = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/zebra/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.elephant = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/elephant/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.amongus = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/amongus/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.space = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/space/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.image);
            })
            .catch(error => {
                rej(error)
            })
    });
}

exports.eight_ball = function(userID, token) {
    return new Promise((resolve,rej) => {
        axios
            .get(`https://bunkapi.xyz/api/v1/8ball/${userID}?token=${token}`)
            .then(res => {
                resolve(res.data.data);
            })
            .catch(error => {
                rej(error)
            })
    });
}
